Classes/git/PBGitGraphLine.h

```diff
@@ -8,8 +8,8 @@
 
 struct PBGitGraphLine
 {
-	int upper      : 1;
-	int from       : 8;
-	int to         : 8;
-	int colorIndex : 8;
+	int upper       : 1;
+	long from       : 8;
+	long to         : 8;
+	int colorIndex  : 8;
```

Comment:
Tracing back, it looks like `long` comes from [`add_line` in `PBGitGrapher`][1], which in turn comes from `newPos` and `currentLanes->size()` in `-[PBGitGrapher decorateCommit]`. 

Since these are bitfields, I don't think changing the type to long buys you anything and will still just truncate larger values silently. Since these bitfields are all exactly 8 bits or 1 bit, they might be better as normal `u/int8_t` fields instead. The total struct size should be the same, and you'll get better type checking.

If we want to keep these as 8 bit fields, then I think `add_line` should be where they get checked (saturate or assert if too big?) and cast down to the smaller type.

[1]: https://github.com/gitx/gitx/pull/16/files#diff-b35e2a8d1d1fb96db59c87a71371690cL49